sql/catalyst/src/main/scala/org/apache/spark/sql/connector/expressions/expressions.scala

```diff
@@ -43,20 +43,22 @@ private[sql] object LogicalExpressions {
   def reference(name: String): NamedReference =
     FieldReference(parser.parseMultipartIdentifier(name))
 
+  def reference(nameParts: Seq[String]): NamedReference = FieldReference(nameParts)
```

Comment:
It would be good to use a different name to distinguish this from the case where the reference name is parsed. What about updating the other one to `parseReference`?

Reply:
good idea!